src/loader.rs

```diff
@@ -0,0 +1,87 @@
+extern crate libloading;
+use self::libloading::{Library, Symbol};
+use super::driver::{Driver, DriverSchema};
+use std::collections::HashMap;
+
+pub type Result<T> = ::std::io::Result<T>;
+
+/// Trait implemented by anthing that wishes to load a driver at runtime.
+pub trait DriverLoader {
+  fn get_driver<'a>(&self) -> Option<&'a Driver>;
+  fn get_schema<'a>(&self) -> Option<&'a DriverSchema>;
+}
+
+pub struct Loader {
+  names: HashMap<String, String>,
+  libs: HashMap<String, Library>,
+  drivers: HashMap<String, Box<DriverLoader>>
+}
+
+impl Loader {
+  pub fn new() -> Self {
+    Loader {
+      names: HashMap::new(),
+      libs: HashMap::new(),
+      drivers: HashMap::new()
+    }
+  }
+
+  fn get_loc(&self, name: String) -> String {
+    self.names.get(&name).unwrap().clone()
```

Comment:
I can guarantee that it is only called with something that was loaded (Maybe, being re-written anyways)